### ISSUE TITLE
Add configuration option to specify which dll's to enumerate when using .NET enumerations as datasource

### ISSUE THREAD

**Issue by davidstroem** [leekelleher/umbraco-contentment#75]
When using .NET enums as data source in a DataList, Contentment enumerates all DLLs in the /bin folder when finding all defined .NET enums.

This can lead to the editor / admin seeing a huge list of .NET enums. In our project it seems to also lead to an error because of DLLs not being the correct version. Although it is not clear to me why we have this error.

I would like to be able to add the names of the DLLs to be enumerated somewhere in a special config section. This way we could tell Contentment to only look in certain DLLs. 

It would also be useful to be able to specify wildcards.

Something like this:

```xml
<contentment>
  <scanAssembly>
    <add assembly="*" />
    <remove assembly="MyAssembly" />
    <remove assembly="MyAssembly2" />
  </scanAssembly>
</contentment>
```

**Comment by davidstroem** [comment]
Hi @leekelleher 
Thank you for replying. I totally understand that you do not have time to solve this. Let me know if you have any thoughts on have to solve this. I will see if I can find the time to try and add some code to a local repo, and maybe we can work something out.
And then I will look at creating my own custom data source.

**Comment by davidstroem** [comment]
This is from the console in dev tools in chromium. I am not sure why it wants a specific version of the dll and other dll's.
Oh, and sorry, it is in danish.
![image](https://user-images.githubusercontent.com/80402985/110653391-7d8b1100-81bd-11eb-9bfe-045ef6d0d05e.png)

**Comment by leekelleher** [comment]
In case it helps with making a custom data-source, feel free to take what you need from the Enum data-source code. <https://github.com/leekelleher/umbraco-contentment/blob/1.2.1/src/Umbraco.Community.Contentment/DataEditors/DataList/DataSources/EnumDataListSource.cs#L66>

---

Thanks for the console error, I'll take a look. 👍

**Comment by leekelleher** [comment]
Closing off this ticket.  I took a look at the error, I wasn't able to reproduce the issue, (sorry).

Another alternate workaround suggestion would be the possibility to use an AngularJS interceptor to hijack the WebAPI request, re-route it to your own API controller endpoint? Hacky, but a workaround (at present).

If at some point in future I do introduce a configuration file, I'll re-consider the original suggestion. 👍
